### PR TITLE
Rename starter name from azure-starter-* into starter-azure-*

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

**Comment by xscript** [comment]
We also need to update the titles and terms in all `README` files, right?

**Comment by xscript** [comment]
Accidentally clicked on the `close` button. Sorry about that. 😂

File: spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ Maven coordinates:
 ----
 <dependency>
     <groupId>com.microsoft.azure</groupId>
-    <artifactId>spring-azure-starter-cache</artifactId>
+    <artifactId>spring-starter-azure-cache</artifactId>
```
[the remark this answers is not in the document]
Comment (warrenzhu25) [reply spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc]: My understanding is that all spring starters are spring boot starter. So no need for boot. But `cache` and `storage` are not spring cloud specific concept, so without `cloud`.

[the remark this answers is not in the document]
Comment (xscript) [reply spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc]: Why spring has the `spring-boot-starter` prefix for all its starters? https://github.com/spring-projects/spring-boot/tree/master/spring-boot-project/spring-boot-starters

[the remark this answers is not in the document]
Comment (xscript) [reply spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc]: My concern is that this kind of discrepancy will make our packages look less professional.

[the remark this answers is not in the document]
Comment (warrenzhu25) [reply spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc]: Another motivation behind this is that `spring-boot-starter` name has been occupied by azure spring boot project. We need a way to differentiate. And this naming is decided by team.

[the remark this answers is not in the document]
Comment (xscript) [reply spring-cloud-azure-samples/spring-cloud-azure-cache-sample/README.adoc]: I created #314 to track this issue separately. Let's merge this PR first.

**Pull request merged**